validate/error.go

```diff
@@ -13,22 +14,81 @@ const referenceTemplate = "https://github.com/opencontainers/runtime-spec/blob/v
 type ErrorCode int
 
 const (
+	// NonError represents that an input is not an error
+	NonError ErrorCode = iota
+	// NonRFCError represents that an error is not a rfc2119 error
+	NonRFCError
+
+	// ConfigFileExistence represents the error code of 'config.json' existence test.
+	ConfigFileExistence
+	// ArtifactsInSignleDir represents the error code of artifacts place test.
+	ArtifactsInSingleDir
+
+	// SpecVersion represents the error code of specfication version test.
+	SpecVersion
+
+	// RootOnNonHyperV represents the error code of root setting test on non hyper-v containers
+	RootOnNonHyperV
+	// RootOnNonHyperV represents the error code of root setting test on hyper-v containers
+	RootOnHyperV
+	// PathFormatOnwindows represents the error code of the path format test on Window
+	PathFormatOnWindows
+	// PathName represents the error code of the path name test
+	PathName
+	// PathExistence represents the error code of the path existence test
+	PathExistence
+	//  ReadonlyFilesystem represents the error code of readonly test
+	ReadonlyFilesystem
+	// ReadonlyOnWindows represents the error code of readonly setting test on Windows
+	ReadonlyOnWindows
+
 	// DefaultFilesystems represents the error code of default filesystems test.
-	DefaultFilesystems ErrorCode = iota
+	DefaultFilesystems
 )
 
 type errorTemplate struct {
 	Level     rfc2119.Level
 	Reference func(version string) (reference string, err error)
 }
 
+var (
+	containerFormatRef = func(version string) (reference string, err error) {
+		return fmt.Sprintf(referenceTemplate, version, "bundle.md#container-format"), nil
+	}
+	specVersionRef = func(version string) (reference string, err error) {
+		return fmt.Sprintf(referenceTemplate, version, "config.md#specification-version"), nil
+	}
+	rootRef = func(version string) (reference string, err error) {
+		return fmt.Sprintf(referenceTemplate, version, "config.md#root"), nil
+	}
+	defaultFSRef = func(version string) (reference string, err error) {
+		return fmt.Sprintf(referenceTemplate, version, "config-linux.md#default-filesystems"), nil
+	}
+)
+
 var ociErrors = map[ErrorCode]errorTemplate{
-	DefaultFilesystems: errorTemplate{
-		Level: rfc2119.Should,
-		Reference: func(version string) (reference string, err error) {
-			return fmt.Sprintf(referenceTemplate, version, "config-linux.md#default-filesystems"), nil
-		},
-	},
+	// NonRFCError represents that an error is not a rfc2119 error
```

Comment:
I don't think we need to duplicate this comment here (you already have it on line 19).